int64 pulse_len
---
int64 ctrl_mode
